tests/apps/test_graph_utils.py

```diff
@@ -193,42 +193,42 @@ class TestIsClique:
     """Tests for the function `strawberryfields.apps.graph.utils.is_clique` """
 
     def test_no_false_negatives(self, dim):
-        """ Tests that cliques are labelled as such"""
+        """Tests that cliques are labelled as such"""
         g = nx.complete_graph(dim)
         assert utils.is_clique(g)
 
     def test_no_false_positives(self, dim):
-        """ Tests that non-cliques are labelled as such"""
+        """Tests that non-cliques are labelled as such"""
         g = nx.empty_graph(dim)
         assert not utils.is_clique(g)
 
 
 @pytest.mark.parametrize("dim", range(2, 10))
 class TestC0:
-    """ Tests function c_0 that generates the set of nodes connected to all nodes in a clique"""
-
-    def test_grows_to_clique(self, dim):
-        """ Tests that adding nodes from C0 to a clique results in another clique"""
-        A = nx.complete_graph(dim)
-        S = [0]
-        K = utils.c_0(S, A)
-        while K:
-            new_node = K[0]
-            S.append(new_node)
-            K = utils.c_0(S, A)
-
-        assert utils.is_clique(A.subgraph(S))
-
-    def test_c0_comp_graph(self, dim):
-        """ Tests that the set C0 for a node in a clique consists of all remaining nodes"""
+    """Tests function :math:`c_0` that generates the set of nodes connected to all nodes in a
+    clique"""
+
+    @pytest.mark.parametrize("subdim", range(2, 10))
+    def test_correct_c_0(self, dim, subdim):
+        """Tests that :math:`c_0` is generated correctly for the lollipop graph"""
+        g = nx.lollipop_graph(dim, 1)
+        s = set(range(subdim))
+        res = set(utils.c_0(s, g))
+
+        assert res not in s
+        assert {dim + 1} not in res
+        assert res | s == set(range(dim))
+
+    def test_c_0_comp_graph(self, dim):
+        """ Tests that the set :math:`c_0` for a node in a clique consists of all remaining nodes"""
         A = nx.complete_graph(dim)
         S = [dim - 1]
         K = utils.c_0(S, A)
 
         assert K == list(range(dim - 1))
 
-    def test_c0input_is_clique(self, dim):
-        """ Tests if function raises a ``ValueError`` when input is not a clique"""
+    def test_c_0_input_is_clique(self, dim):
```

Comment:
Suggest making the name of the test more explicitly reflect the test, e.g.
`test_c_0_error_on_not_clique_input`